android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/RibCoroutineWorker.kt

```diff
@@ -198,25 +198,27 @@ public fun RibCoroutineWorker.asWorker(
 internal open class WorkerToRibCoroutineWorkerAdapter(private val worker: Worker) :
   RibCoroutineWorker {
   override suspend fun onStart(scope: CoroutineScope) {
-    withContext(worker.coroutineDispatcher) { worker.onStart(scope.asWorkerScopeProvider()) }
+    val dispatcher = worker.coroutineDispatcher ?: RibDispatchers.Unconfined
+    withContext(dispatcher) { worker.onStart(scope.asWorkerScopeProvider()) }
   }
 
   override fun onStop(cause: Throwable): Unit = worker.onStop()
 }
 
 internal open class RibCoroutineWorkerToWorkerAdapter(
   private val ribCoroutineWorker: RibCoroutineWorker,
-  private val dispatcher: CoroutineDispatcher?,
+  dispatcher: CoroutineDispatcher?,
```

Comment:
We can just use `override val coroutineDispatcher` in constructor instead

Reply:
Nice catch, will update